src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/MarkupCompiler.cs

```diff
@@ -762,6 +762,13 @@ static void ThrowCompilerExceptionImpl(string message)
 
         internal void OnError(Exception e)
         {
+            // Don't treat an AssemblyVersion parsing error as a XamlParseException.
+            // Throw it back to the task execution.
+            if(e is AssemblyVersionParseException)
+            {
+                throw e;
```

Comment:
Should this be `System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e).Throw();` in order to add to the stack trace instead of overwriting it?

Reply:
Yeah should just be `throw`.

Reply:
Or rather, just throw in the XamlParser.